packages/hint-scoped-svg-styles/src/hint.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @fileoverview Scoped SVG Styles checks if SVG styles affect any other elements outside the svg.
+ */
+
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, ProblemLocation } from 'hint/dist/src/lib/types';
+import { debug as d, HTMLElement } from '@hint/utils';
+
+import { StyleEvents, StyleParse } from '@hint/parser-css';
+import { getCSSCodeSnippet } from '@hint/utils/dist/src/report/get-css-code-snippet';
+import { Rule } from 'postcss';
+
+import meta from './meta';
+import { getMessage } from './i18n.import';
+
+const debug: debug.IDebugger = d(__filename);
+
+const findParentSVGElement = (element: HTMLElement): HTMLElement | null => {
+    if (!element.parent) {
+        return null;
+    }
+
+    if (element.parent.nodeName === 'svg') {
+        return element.parent;
+    }
+
+    return findParentSVGElement(element.parent);
+};
+
+const isOutsideParentSVG = (parentSVG: HTMLElement) => {
+    return (element: HTMLElement): boolean => {
+        const elementsParentSVG = findParentSVGElement(element);
+
+        if (!elementsParentSVG) {
+            return true;
+        }
+        if (!elementsParentSVG.isSame(parentSVG)) {
+            return true;
+        }
+
+        return false;
+    };
+};
+
+const getLocation = (rule: Rule): ProblemLocation | null => {
+    const start = rule.source && rule.source.start;
+
+    if (start) {
+        return {
+            column: start.column - 1,
+            line: start.line - 1
+        };
+    }
+
+    return null;
+};
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ScopedSvgStylesHint implements IHint {
+    public static readonly meta = meta;
+
+    public constructor(context: HintContext<StyleEvents>) {
+        /** Generate a report message from elements matched outside the SVG. */
+        const formatRuleMessage = (
```

Comment:
move the function parameter to this line:
`const formatRuleMessage = (numberOfElementsOutsideSVG: number): string => {`

Reply:
done